Fix: Align Document Outline padding in List View sidebar to match surrounding panels (#73576)

manhphuc opened the pull request:
## What?

This PR fixes inconsistent padding inside the Outline panel in the List View sidebar.  
Closes: #73576

Previously, the `.document-outline` area had no padding, causing the content to touch the left/right edges while the statistics block above (“Characters / Words / Time to read”) had proper spacing. This created a visual imbalance.

## Why?

To improve UI consistency and readability.  
The Outline panel is part of the same sidebar group, so its spacing should visually match the panels above it.

## How?

Updated SCSS in:

`packages/editor/src/components/list-view-sidebar/style.scss`

Changes include:

- Restored the original List View panel padding (`$grid-unit-05`) so its spacing matches trunk.
- Applied the new padding (`$grid-unit-20`) only to `.document-outline`.
- Overrode the global top margin on `.document-outline` (`margin-top: 0`) so it doesn’t stack with the new padding.

This ensures consistent spacing in the Outline panel without affecting the List View panel layout.


## Testing Instructions

1. Open the block editor.
2. Open List View → Outline tab.
3. Check that:
   - The Outline list has padding on all sides.
   - It visually aligns with the “Characters / Words / Time to read” stats panel.
4. Scroll inside the Outline panel to confirm scrollbar behavior remains correct.
5. Resize the editor window and ensure no layout shift or regression.

## Keyboard Testing

1. Use Tab to move focus into List View.
2. Navigate into the Outline panel with arrow keys.
3. Ensure:
   - The focus outline is fully visible (not clipped).
   - Navigation order is unchanged.
   - Padding does not interfere with focus movement.

## Screenshots

| Before | After |
|--------|-------|
| ![Before](https://github.com/user-attachments/assets/38d87ae2-fb95-4442-b627-04c5638eb539) | ![After](https://github.com/user-attachments/assets/6cd4b87f-fee6-4d8f-ae3e-69e40a32b626) |

manhphuc committed:
Fix: Adjust padding for Document Outline in List View sidebar to match stats section (#73576)

Fix: Adjust padding for Document Outline in List View sidebar to match stats section (#73576)

This patch ensures consistent spacing between the List View panel and the Document Outline panel in the sidebar.

- Keep the original List View panel padding (`$grid-unit-05`) to match trunk.
- Apply the new padding (`$grid-unit-20`) only to the Document Outline section.
- Reset the top margin on `.document-outline__summary` to prevent stacked margin + padding.

Both panels have been re-tested locally and now match the expected spacing.

juanfra reviewed (changes requested):
Thanks so much for the PR! 

I did notice a couple of things to double-check. The changes are also affecting the spacing of the list view itself (`.editor-list-view-sidebar__list-view-panel-content`), which we shouldn't modify.

Separately, the vertical spacing (top) for the Overview ends up being padding + margin, adding up to 36px.

https://github.com/user-attachments/assets/2adedbb9-3bd1-4392-be7f-3c6fe180f721

manhphuc committed:
Fix: Override global margin for Document Outline to prevent margin + padding stacking

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,21 @@
+	// Override the global top margin (20px) so it doesn’t stack with the new padding.
+	margin-top: 0;
+	padding: $grid-unit-20;
+}
```
We may want to do this directly in the document [outline component](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/document-outline/style.scss#L5). We may even consider deleting the margin property. I can't think of reasons to override the styles only in the list view.

Are you using the [child combinator](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/Child_combinator) (`>`) for any reason in particular here?

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,21 @@
 	padding: $grid-unit-05;
 }
 
+// Apply the new padding only to the Document Outline panel.
```
These comments will be hard to interpret outside the context of this PR. Once this is merged, "new" and "old" padding won't be meaningful references anymore, so we should avoid adding this level of detail in the source. In general, unless there’s something truly non-obvious, we try to keep CSS comments minimal.

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,21 @@
 
-	// The table cells use an extra pixels of space left and right. We compensate for that here.
+// The table cells use an extra pixels of space left and right. We compensate for that here.
+.editor-list-view-sidebar__list-view-panel-content {
+	// Keep the original padding for the List View panel so its spacing matches trunk.
 	padding: $grid-unit-05;
 }
```
Same thing I mentioned [here](https://github.com/WordPress/gutenberg/pull/73595/files#r2588394107).

juanfra reviewed (changes requested):
Thanks again for the work on this! 

I think the approach to the CSS could use another pass. We should double-check whether the outline component is used in places other than the List View sidebar, and adjust the selectors so we're targeting the correct context without introducing unintended changes elsewhere.

manhphuc committed:
Editor: Align Outline panel spacing with surrounding panels (fix #73576)

juanfra commented on packages/editor/src/components/document-outline/style.scss:
```diff
@@ -2,7 +2,7 @@
 
 .document-outline {
-	margin: 20px 0;
+	margin: 0;
```
You can remove this line. The initial value for margin is already 0, so we don’t need to reset it explicitly.

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,19 @@
 
-	// The table cells use an extra pixels of space left and right. We compensate for that here.
+// The table cells use a few extra pixels of horizontal space.
+// Apply compensating padding for the List View panel.
```
Let’s keep the existing comment as-is. It already provides the right level of context.

juanfra reviewed (changes requested):
Thanks again for the persistence on this one. Appreciate you sticking with it. 

There are just a couple of small adjustments needed

manhphuc committed:
Fix: Adjust document outline spacing and keep existing comment context as suggested

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,17 @@
+}
 
-	// The table cells use an extra pixels of space left and right. We compensate for that here.
+.editor-list-view-sidebar__list-view-panel-content {
```
Let's please keep the comment.

```suggestion
// The table cells use an extra pixels of space left and right. We compensate for that here.
.editor-list-view-sidebar__list-view-panel-content {
```

juanfra commented on packages/editor/src/components/list-view-sidebar/style.scss:
```diff
@@ -22,11 +22,17 @@
+// Match the Outline panel’s padding with the surrounding panels.
+.editor-list-view-sidebar__list-view-container > .document-outline {
+	padding: $grid-unit-20;
+}
```
Let's nest this inside of `.editor-list-view-sidebar__list-view-container` so that's better organized, we already have that in place.

The change should look something like this:

```
.editor-list-view-sidebar__list-view-container {
	// rules that are already there
 	> .document-outline {
		// rules you added here
	}
}
```

manhphuc replied:
Done! I’ve kept the comment and nested the outline styles as suggested. Thanks for the helpful review 😊

juanfra reviewed (changes requested):
Nice work, thanks for your persistence. A few small changes and we should be ready to go.

manhphuc committed:
Fix list view sidebar padding and SCSS organization

- Preserve explanatory comment for table cell padding
- Nest document outline styles inside list view container

juanfra reviewed (approved):
Nice work on this! Appreciate your persistence in getting this across the finish line.
